src/DataGrid.tsx

```diff
@@ -525,18 +527,22 @@ function DataGrid<R, SR, K extends Key>(
 
   function handleKeyDown(event: KeyboardEvent<HTMLDivElement>) {
     const { idx, rowIdx, mode } = selectedPosition;
+    const column = columns[idx];
+
     if (mode === 'EDIT') return;
 
-    if (onCellKeyDown && isRowIdxWithinViewportBounds(rowIdx)) {
```

Comment:
why do we need to check `isRowIdxWithinViewportBounds` before call `onCellKeyDown` ?

Reply:
I think because row would be undefined otherwise. Same for summary row. We may have to add a type `HEADER | ROW | SUMMARY`